File: day02/day02-part1.php
<?php

$input = file(__DIR__ . '/real-input');
$input = array_merge(...array_map(fn ($line) => explode(',', $line), $input));

$answer = 0;

foreach ($input as $line) {
    [$first, $last] = explode('-', $line);

    // For each "number" check if it's made of 2 equal halves of the same number (e.g. 11 = 1 + 1)
    for ($i = (int)$first; $i <= (int)$last; $i++) {
        $str = (string)$i;

        // Must be 2 groups of the same digits, therefore odd lengths can be ignored
        if (strlen($i) % 2 !== 0 ) {
            continue;
        }

        // First half must be same as second half
        $firstHalf = substr($str, 0, strlen($str) / 2);
        $secondHalf = substr($str, strlen($str) / 2);

        if ($firstHalf !== $secondHalf) {
            continue;
        }

        $answer += $i;
    }
}

echo "Answer: $answer\n";

File: day02/day02-part1.py

def readFile():
    with open("python-input") as file:
        return file.read().strip().split(",")

contents = readFile()

ans = 0

for pair in contents:
    first,last = pair.split("-")
    for num in range(int(first),int(last)+1):
        numStr = str(num)
        numLen = len(numStr)

        if numLen % 2 == 1:
            continue

        # split number in half, check if first half equals second half
        firstHalf, secondHalf = numStr[0:int(numLen/2)], numStr[int(numLen/2):]

        if firstHalf == secondHalf:
            ans += num

print(ans)
